ssl/statem/statem_srvr.c

```diff
@@ -1956,9 +1987,9 @@ WORK_STATE tls_post_process_client_hello(SSL *s, WORK_STATE wst)
         wst = WORK_MORE_B;
     }
     if (wst == WORK_MORE_B) {
-        if (!s->hit || s->hello_retry_request) {
+        if (!s->hit || SSL_IS_TLS13(s)) {
```

Comment:
This one seems to just be about not taking the "else" branch, as everything inside this branch is either excluded if SSL_IS_TLS13() or requires !s->hit.
But sure, it makes sense now -- thanks for the explanation.